src/test/compile-fail/empty-struct-unit-expr.rs

```diff
@@ -24,10 +24,10 @@ enum E {
 fn main() {
     let e2 = Empty2(); //~ ERROR expected function, found `Empty2`
     let e4 = E::Empty4();
-    //~^ ERROR `E::Empty4` is being called, but it is not a function
+    //~^ ERROR expected a function, but `E::Empty4` is `E`
```

Comment:
Furthermore, #39727, which this PR is supposed to fix, is not about enum variants at all.

(Statistically, "declared without arguments" still probably a better guess if the callee is indeed a unit variant.)